test/jasmine/tests/select_test.js

```diff
@@ -109,6 +123,630 @@ var BOXEVENTS = [1, 2, 1];
+            testCase('choropleth', require('@mocks/geo_choropleth-text.json'), 440, 163, [6]),
+            testCase('scattermapbox', require('@mocks/mapbox_0.json'), 650, 195, [[2], []], {},
+              { mapboxAccessToken: require('@build/credentials.json').MAPBOX_ACCESS_TOKEN }),
+            testCase('scattergeo', require('@mocks/geo_scattergeo-locations.json'), 285, 240, [1]),
+            testCase('scatterternary', require('@mocks/ternary_markers.json'), 485, 335, [7]),
+
+            // Note that first trace (carpet) in mock doesn't support selection,
+            // thus undefined is expected
+            testCase('scattercarpet', require('@mocks/scattercarpet.json'), 532, 178,
+              [undefined, [], [], [], [], [], [2]], { width: 1100, height: 450 }),
+
+            // scatterpolar and scatterpolargl do not support pan (the default),
+            // so set dragmode to zoom
+            testCase('scatterpolar', require('@mocks/polar_scatter.json'), 130, 290,
+              [[], [], [], [19], [], []], { dragmode: 'zoom' }),
+            testCase('scatterpolargl', require('@mocks/glpolar_scatter.json'), 130, 290,
+              [[], [], [], [19], [], []], { dragmode: 'zoom' }).enableGl(),
+            testCase('splom', require('@mocks/splom_lower.json'), 427, 400, [[], [7], []]).enableGl()
+        ]
+          .forEach(function(testCase) {
+              var ciAnnotation = testCase.gl ? 'gl' : 'flaky';
+              it('@' + ciAnnotation + ' trace type ' + testCase.label, function(done) {
+                  Plotly.plot(gd, testCase.mock.data, testCase.mock.layout, testCase.mock.config)
+                    .then(function() {
+                        return _immediateClickPt(testCase);
+                    })
+                    .then(function() {
+                        assertSelectedPoints(testCase.expectedPts);
+                        return Plotly.relayout(gd, 'dragmode', 'lasso');
+                    })
+                    .then(function() {
+                        _clickPt(testCase);
+                        return deselectPromise;
+                    })
+                    .then(function() {
+                        assertSelectionCleared();
+                        return _clickPt(testCase);
+                    })
+                    .then(function() {
+                        assertSelectedPoints(testCase.expectedPts);
+                    })
+                    .catch(failTest)
+                    .then(done);
+              });
+          });
+    });
+
+    describe('triggers \'plotly_selected\' before \'plotly_click\'', function() {
+        [
+            testCase('cartesian', require('@mocks/14.json'), 270, 160, [7]),
+            testCase('geo', require('@mocks/geo_scattergeo-locations.json'), 285, 240, [1]),
+            testCase('ternary', require('@mocks/ternary_markers.json'), 485, 335, [7]),
+            testCase('mapbox', require('@mocks/mapbox_0.json'), 650, 195, [[2], []], {},
```

Comment:
We usually skip (with tag `@noCI`) mapbox tests on CI as they tend to be too resource-intensive. 

@rmoestl do you recall any of your newly added mapbox test failing on CI?

Reply:
No, I don't. Anyways, I've separated the (new) mapbox tests to use the `@noCI` annotation.

Reply:
Great. Thank you!